hoodie-client/src/main/java/com/uber/hoodie/io/HoodiePartitionHelper.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.io;
+
+import com.uber.hoodie.common.model.HoodieRecord;
+import com.uber.hoodie.table.WorkloadProfile;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import scala.Tuple2;
+import scala.Tuple3;
+
+/**
+ * HoodiePartitionHelper is responsible for splitting Hoodie upsert stage into smaller spark
+ * stages.
+ */
+public class HoodiePartitionHelper {
```

Comment:
Probably should not be public, we don't want clients to end up using such a helper ?